spark/src/test/scala/org/apache/spark/sql/delta/DeltaInsertIntoTest.scala

```diff
@@ -0,0 +1,272 @@
+    override val mode: SaveMode = SaveMode.Append
+    val name: String = s"Streaming toTable()"
+    def runInsert(columns: Seq[String], whereCol: String, whereValue: Int): Unit = {
+      val tablePath = DeltaLog.forTable(spark, TableIdentifier("target")).dataPath
+      val query = spark.readStream
+        .table("source")
+        .writeStream
+        .option("checkpointLocation", tablePath.toString)
+        .format("delta")
+        .toTable("target")
+      query.processAllAvailable()
+    }
+  }
+
+  /** Collects all the types of insert previously defined. */
+  protected lazy val allInsertTypes: Seq[Insert] = Seq(
+        SQLInsertOverwriteReplaceWhere,
+        SQLInsertOverwritePartitionByPosition,
+        SQLInsertOverwritePartitionColList,
+        DFv2Append,
+        DFv2Overwrite,
+        DFv2OverwritePartition,
+        StreamingInsert
+  ) ++ (for {
+      mode: SaveMode <- Seq(SaveMode.Append, SaveMode.Overwrite)
+      insert: Insert <- Seq(
+        SQLInsertByPosition(mode),
+        SQLInsertColList(mode),
+        SQLInsertByName(mode),
+        DFv1InsertInto(mode),
+        DFv1SaveAsTable(mode)
+      )
+    } yield insert)
+
+  // scalastyle:off argcount
+  /**
+   * Test runner to cover INSERT operations defined above.
+   * @param name             Test name
+   * @param initialSchemaDDL Initial schema of the table to be inserted into (as a DDL string).
+   * @param initialJsonData  Initial data present in the table to be inserted into (as a JSON
+   *                         string).
+   * @param partitionBy      Partition columns for the initial table.
+   * @param insertSchemaDDL  Schema of the data to be inserted (as a DDL string).
+   * @param insertJsonData   Data to be inserted (as a JSON string)
+   * @param overwriteWhere   Where clause for overwrite PARTITION / REPLACE WHERE (as
+   *                         colName -> value)
+   * @param expectedSchema   Expected schema of the table after the insert. Only set this parameter
+   *                         if the insert is expected to succeed.
+   * @param checkError       A check to run on the exception thrown by the insert operation. Only
+   *                         set this parameter if the insert is expected to fail.
+   * @param includeInserts   List of insert types to run the test with. Defaults to all inserts.
+   * @param excludeInserts   List of insert types to exclude when running the test. Defaults to no
+   *                         inserts excluded.
+   * @param confs            Custom spark confs to set before running the insert operation.
+   */
+  def testInserts(name: String)(
+      initialSchemaDDL: String,
+      initialJsonData: Seq[String],
+      partitionBy: Seq[String] = Seq.empty,
+      insertSchemaDDL: String,
+      insertJsonData: Seq[String],
+      overwriteWhere: (String, Int),
+      expectedSchema: StructType = null,
+      checkError: SparkThrowable => Unit = null,
```

Comment:
We prefer `Option` over `null`s in Scala. In this case, it sounds like we want `Either`: https://www.scala-lang.org/api/2.13.6/scala/util/Either.html